File: backend/Dtos/response/PacienteResponseDTO.cs
namespace backend.Dtos.response;

public record PacienteResponseDTO(
    int Id_Paciente,
    string Nombre,
    string Apellido,
    string Dni,
    string Email,
    string? Telefono,
    string? Genero,
    decimal Peso_Inicial,
    decimal Altura_Cm,
    string Estado,
    string? TokenAcceso,
    string? CodigoAcceso,
    decimal PesoActual,
    decimal Imc,
    ObjetivoResponseDTO? Objetivo, 
    List<PatologiaResponseDTO> Patologias,

    List<ArchivoResponseDTO>? ArchivosNutricionista = null,
    List<PesajeResponseDTO>? HistorialPeso = null,
    DietaResponseDTO? DietaActual = null
);

File: frontend/src/types/dto/response/PacienteResponseDTO.ts
import type { ArchivoResponseDTO } from "./ArchivoResponseDTO";
import type { DietaResponseDTO } from "./DietaResponseDTO";
import type { ObjetivoResponseDTO } from "./ObjetivoResponseDTO";
import type { PatologiaResponseDTO } from "./PatologiaResponseDTO";
import type { PesajeResponseDTO } from "./PesajeResponseDTO";

export interface PacienteResponseDTO {
    Id_Paciente: number;
    Nombre: string;
    Apellido: string;
    Dni: string;
    Email: string;
    Telefono?: string;
    Genero?: string;
    Peso_Inicial: number;
    Altura_Cm: number;
    Estado: string;
    TokenAcceso?: string;
    CodigoAcceso?: string;
    
    PesoActual: number;
    Imc: number;
    
    Objetivo?: ObjetivoResponseDTO;
    Patologias: PatologiaResponseDTO[];

    ArchivosNutricionista: ArchivoResponseDTO[];
    HistorialPeso: PesajeResponseDTO[];
    DietaActual?: DietaResponseDTO;
}